dask/utils.py

```diff
@@ -763,6 +754,15 @@ def dispatch(self, cls):
                 register()
                 self._lazy.pop(toplevel, None)
                 return self.dispatch(cls)  # recurse
+            try:
```

Comment:
Why are we switching the order here? What went wrong before?

Reply:
https://github.com/dask/dask/pull/10808#discussion_r1455638967